nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -0,0 +1,142 @@
+package com.uber.nullaway;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Attribute;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.HashSet;
+import java.util.List;
+
+public class GenericsChecks {
+
+  // check that type is a valid instantiation of its generic type
+  public static void checkInstantiatedType(
+      Type type, VisitorState state, Tree tree, NullAway analysis, Config config) {
+    CodeAnnotationInfo codeAnnotationInfo = CodeAnnotationInfo.instance(state.context);
+    if (config.isJSpecifyMode()) {
+      // typeArguments used in the instantiated type (like for Foo<String,Integer>, this gets
+      // [String,Integer])
+      // if base type is unannotated do not check for generics
+      // temporary check to handle testMapComputeIfAbsent
+      if (codeAnnotationInfo.isSymbolUnannotated(type.tsym, config)) {
+        return;
+      }
+
+      com.sun.tools.javac.util.List<Type> typeArguments = type.getTypeArguments();
+      HashSet<Integer> nullableTypeArguments = new HashSet<Integer>();
+      int index = 0;
```

Comment:
I think it would be clearer to do `for (int index = 0; index < typeArguments.size(); index++)` and then use `get(index)` to get the current type argument.  An enhanced for loop is best when you don't care about the index values